### DISCUSSION TITLE
zipWith?

### DISCUSSION THREAD

**Discussion by hillin** [iter-tools/iter-tools#482] [Ideas]
Currently `zip` cannot be used in a pipeline, not sure if it's intended, or yet another typing problem:

```typescript
// does not compile
execPipe(
  [1,2,3],
  zip([4,5,6])
);
```

or should we have a `zipWith` operator for this role?

**Comment by conartist6** [comment]
No, we will not create a `zipWith` operator for that use case. For now you can write:
```js
execPipe(
  [1,2,3],
  iter => zip([4,5,6], iter)
);
```
I don't have any problem at all with it being necessary use an arrow function as a pipe argument. Part of the reason for this is that the `execPipe` is just a transitional measure while we wait for the language to accept a specification for a pipe operator of its own. When the pipeline operator is released you'll write that as:
```js
return [1, 2, 3]
    |> zip([4, 5, 6], %)
```

**Reply by hillin** [reply]
Yes the language built-in piping would make many things easier and more elegant. I myself is a F# guy tho so I prefer the unary piping proposal, but either would be a great thing for heavy rx and iter-tools user like me.
